### DISCUSSION TITLE
Is it possible to access the jpa3 beans?

### DISCUSSION THREAD

**Discussion by fschon** [opencrx/opencrx#137] [Q&A]
I would like to java serialise the jpa3 bean, for example the org.opencrx.kernel.account1.jpa3.Account bean. 

1) Is it possible to access these beans and not the cci/jmi Objects
2) Are the jpa3 beans ever actually instantiated in the opencrx/mdx framework?
3) And if so, where are they used in the framework, i.e. is there a class responsible to taking the jpa3 beans and using the associated jdo xml mapping to convert it to Records?

Thanks for any pointers.

**Comment by fschon** [comment]
I figured out that it is possible to retrieve the jpa3 bean by using pm.detatchCopy(). This is a great first step. I only now noticed that any relations to other objects are not coded into the jpa3 beans, so for example for the org.opencrx.kernel.account1.jpa3.Account bean the Address list is not modeled.

`  public <T extends org.opencrx.kernel.account1.cci2.AccountAddress> org.opencrx.kernel.account1.cci2.AccountHasAccountAddress.Address<T> getAddress(
  ){
    throw new javax.jdo.JDOFatalUserException(
      "This signature is not handled by data object",
      new UnsupportedOperationException("This signature is not handled by data object. Use query on composites."),
      this
    );
  }`

How are these relations managed on the java jpa3 side? 
How could I detach the Account object together with the Addresses?

Eventually I would like to detach/retrieve the entire java object graph into POJOs with relationships. Would I have to modify the openmdx code that generates these jpa3 beans to include the relationships, and then modify the detach to pull the whole relationship graph as well?

Appreciate any help.

**Comment by DlRTY-HARRY** [comment]
It is planned that openMDX JPA objects support relationships from openMDX 5 on, which is expected by mid 2025.
